File: C_Programming/Practise_Programs/PP_144.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_144
//
//  Description       : This program counts the number of 1's in the
//                      binary representation of a given decimal number.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 18/08/2026
//
//  Time Complexity   : O(log n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

int main()
{
    int iNo = 0, iCount = 0, iDigit = 0;

    printf("Enter Number : \n");
    scanf("%d", &iNo);

    while(iNo != 0)
    {
        iDigit = iNo % 2;

        if(iDigit == 1)
        {
            iCount++;
        }

        iNo = iNo / 2;
    }

    printf("Number of 1's are : %d\n", iCount);   
    
    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      10
//
//  Sample Output :
//
//      Enter Number :
//      Number of 1's are : 2
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_144.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_144
//
//  Description       : This program prints a right-aligned
//                      decreasing star pattern using nested
//                      for loops. The number of stars decreases
//                      by one in each successive row.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 31/08/2026
//
//  Time Complexity   : O(n^2)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

class PP_144
{
    public static void main(String args[])
    {
        for(int i = 1; i <= 5; i++)
        {
            for(int j = 0; j <= i; j++)
            {
                System.out.print(" ");
            }

            for(int j = 1; j <= 6 - i; j++)
            {
                System.out.print("* ");
            }

            System.out.println();
        }
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//       * * * * *
//        * * * *
//         * * *
//          * *
//           *
//
//////////////////////////////////////////////////////////////////